From 3da6b4f0c1d5699a424a919baf27e91bda10fd46 Mon Sep 17 00:00:00 2001
From: Thomas Jarosch <thomas.jarosch@intra2net.com>
Date: Fri, 26 Aug 2011 11:40:26 +0200
Subject: [PATCH 1/2] Fix file descriptor leak on error

We were not closing the correct file descriptor
if unlink() or init_write_buffer() failed.

Detected by "cppcheck".

Signed-off-by: Thomas Jarosch <thomas.jarosch@intra2net.com>
---
 imap/squat_build.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/imap/squat_build.c b/imap/squat_build.c
index df0f62b..69fc726 100644
--- a/imap/squat_build.c
+++ b/imap/squat_build.c
@@ -552,7 +552,7 @@ static int init_write_buffer_to_temp(SquatIndex *index,
     return SQUAT_OK;
 
 cleanup_fd:
-    close(b->fd);
+    close(fd);
     free(tmp_path);
     return SQUAT_ERR;
 }
-- 
1.7.4.4

